_posts/2022-05-31-scala-on-app-service.md

```diff
@@ -0,0 +1,131 @@
+---
+title: "Scala on App Service"
+toc: true
+toc_sticky: true
+author_name: Denver Brittain
+excerpt: "Run a Scala App using Java SE on App Service"
+category: java
+---
+
+## Prerequisites
+
+This sample is based off of the [Play Framework Hello World Tutorial](https://github.com/playframework/play-samples/tree/2.8.x/play-scala-hello-world-tutorial). 
+
+To follow the steps in this tutorial, you will need the correct version of Java, sbt, and (optionally) Maven or the Azure CLI.
+
+* Java 11
+* sbt 1.3.4 or higher. Note: if you downloaded this project as a zip file from <https://developer.lightbend.com>, the file includes an sbt distribution for your convenience.
+* Maven (Optional)
+* Azure CLI (Optional)
+
+To check your Java version, enter the following in a command window:
+
+```bash
+java -version
+```
+
+To check your sbt version, enter the following in a command window:
+
+```bash
+sbt sbtVersion
+```
+
+If you do not have the required versions, follow these links to obtain them:
+
+* [Java SE](http://www.oracle.com/technetwork/java/javase/downloads/index.html)
+* [sbt](http://www.scala-sbt.org/download.html)
+* [maven](https://maven.apache.org/install.html) (Optional)
+* [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli) (Optional)
+
+## Build and Run the Project
+
+This example Play project was created from a seed template. It includes all Play components and an Akka HTTP server. The project is also configured with filters for Cross-Site Request Forgery (CSRF) protection and security headers.
+
+To build and run the project:
+
+1. First clone the [Scala on App Service](https://github.com/Azure-Samples/scala-on-app-service) repository with the following command: `git clone https://github.com/Azure-Samples/scala-on-app-service` 
+
+2. Change directory into the example project directory, for example: `cd scala-on-app-service`
+
+2. Build the project. Enter: `sbt run`. The project builds and starts the embedded HTTP server. Since this downloads libraries and dependencies, the amount of time required depends partly on your connection's speed.
+
+3. After the message `Server started, ...` displays, enter the following URL in a browser: <http://localhost:80>
+
+The Play application responds: `Welcome to the Hello World Scala POC Tutorial!`
+
+
+## Assemble and Test JAR Locally
+
+To build a .jar file executable for a Java 11 runtime using sbt assembly: 
+
+1. From the project root, run: 
+
+    ```bash
+    sbt assembly
+    ```
+
+    This command produces an executable .jar file in the scala-on-app-service/target/scala-2.14 directory.
+
+2. To test the app locally, run the previously created .jar file:  
+
+    ```bash
+    java -jar target/scala-2.13/<project-name>-assembly-<version>.jar
+    ```
+

```

Comment:
Add a "Step 3" to check the app in the browser again

Add a small paragraph that gives the high-level technicalities of the fat JAR. (You explained it in our 1:1 before, like how the "assembly" command includes the whole Scala runtime or something to that effect)